src/main/java/com/alem/todolist/repository/ToDoListRepository.java

```diff
@@ -1,13 +1,19 @@
 package com.alem.todolist.repository;
 
+import com.alem.todolist.exceptions.MissingArgumentException;
 import com.alem.todolist.model.Task;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
+import java.time.Instant;
 import java.util.Optional;
 
 @Repository
 public interface ToDoListRepository extends JpaRepository<Task, Long> {
 
     Optional<Task> findById(Long id);
+
+    @Query(value = "SELECT t FROM tasks t WHERE t.date = ?1", nativeQuery = true)
+    public int countTasksForGivenDate(Instant dt) throws MissingArgumentException;
```

Comment:
This looks better, but won't work for the case that you are trying to achieve. You have a timestamp, so you have to create a query that would match your date. 
Check Postgres docs or maybe an article like these: https://popsql.com/learn-sql/postgresql/how-to-query-date-and-time-in-postgresql.